src/main/java/uk/gov/pay/ledger/event/model/EventDigest.java

```diff
@@ -58,18 +60,28 @@ public static EventDigest fromEventList(List<Event> events) {
                 .min(ZonedDateTime::compareTo)
                 .orElseThrow();
 
+        String parentResourceExternalId = deriveParentResourceExternalId(events);
+
         return new EventDigest(
                 latestEvent.getEventDate(),
                 latestSalientEventType,
                 latestEvent.getResourceType(),
                 latestEvent.getResourceExternalId(),
-                latestEvent.getParentResourceExternalId(),
+                parentResourceExternalId,
                 events.size(),
                 eventPayload,
                 earliestDate
         );
     }
 
+    private static String deriveParentResourceExternalId(List<Event> events) {
+        return events.stream()
+                .filter(event -> isNotEmpty(event.getParentResourceExternalId()))
+                .map(Event::getParentResourceExternalId)
+                .findFirst()
```

Comment:
I think Java streams have a `.distinct()` method that would work on strings like this.

We might want to do some warning logging if there are more than one? We could still use the result of `findFirst()` but just make some noise.

Reply:
To log warning it is forcing to use stream filtering twice and method could look complex. will think it through and raise a separate PR as needed.